File: SignatureRequests/SignatureRequests.Core/Entities/BoxEntity.cs
using BTSuggestions.Core.Entities;
using System;
using System.ComponentModel.DataAnnotations;

namespace SignatureRequests.Core.Entities
{
    public class BoxEntity: BaseEntity
    {
        [Required]
        public int Width { get; set; }

        [Required]
        public int Height { get; set; }

        [Required]
        public int X { get; set; }

        [Required]
        public int Y { get; set; }

        [Required]
        public string Type { get; set; }

        [Required]
        public string SignerType { get; set; }

        [Required]
        public string SignedStatus { get; set; }

        public virtual RequestEntity Request { get; set; }

        public int? RequestId { get; set; }

        public virtual SignatureEntity Signature { get; set; }

        public int? SignatureId { get; set; }

        [Required]
        public virtual FormEntity Form { get; set; }

        public int FormId { get; set; }

        public int PageNumber { get; set; }

        public bool IsModel { get; set; }

        public string Text { get; set; }

        public DateTime? Date { get; set; }

        
    }
}

File: SignatureRequests/SignatureRequests.Webs/Scripts/react/src/Entities/BoxEntity.ts
import RequestEntity from "./RequestEntity";
import SignatureEntity from "./SignatureEntity";
import FormEntity from "./FormEntity";

class BoxEntity {
  id: number;
  width: number;
  height: number;
  x : number;
  y: number;
  type: string;
  signerType: string;
  signedStatus: string;
  requestId: number;
  signature?: SignatureEntity;
  formId: number;
  pageNumber: number;
  isModel: boolean;
  text?: string;
  date?: Date;
  formHeight: number;
  formWidth: number;


  constructor(data: any) {
    
    this.id = data.Id;
    this.width = data.Width;
    this.height = data.Height;
    this.x = data.X;
    this.y = data.Y;
    this.type = data.Type;
    this.signerType = data.SignerType;
    this.signedStatus = data.SignedStatus;
    this.requestId = data.RequestId;
    this.signature = data.Signature;
    this.formId = data.FormId;
    this.pageNumber = data.PageNumber;
    this.isModel = data.IsModel;
    this.text = data.Text;
    this.date = data.Date;
    this.formHeight = data.FormHeight;
    this.formWidth = data.FormWidth;
  }
}

export default BoxEntity;
